Fix maven central deploy

### 📂 Involved File

`.github/workflows/publish.yml`

---

### ❌ Error

```
> Task :dspfparser:publishMavenJavaPublicationToSonatypeRepository FAILED
gradle/actions: Writing build results to /home/runner/work/_temp/.gradle-actions/build-results/__run_3-1756119433590.json

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dspfparser:publishMavenJavaPublicationToSonatypeRepository'.
> Failed to publish publication 'mavenJava' to repository 'sonatype'
```

---

### 🔎 Evaluation

* The failure occurs in the `dspfparser` module, but **all modules must be published**, so the issue affects the entire project.
* The error seems related to **Sonatype OSSRH deprecation and migration**.
* I received the following notification about a breaking change:

---

### 📢 Migration Notice (from Sonatype / Central Team)

> **Context:** OSSRH will reach end of life on **June 30, 2025**.
> OSSRH users must migrate their namespaces to the **Central Portal** before this date.
>
> Instructions for self-migration:
> [[What is different between Central Portal and legacy OSSRH?](https://central.sonatype.org/faq/what-is-different-between-central-portal-and-legacy-ossrh/#self-service-migration)](https://central.sonatype.org/faq/what-is-different-between-central-portal-and-legacy-ossrh/#self-service-migration)
>
> If you have already migrated, this does not apply. Otherwise:
>
> * Sonatype will **automatically migrate inactive publishers**.
> * To avoid disruption, it is strongly recommended to **migrate manually** before the deadline.
>
> — The Central Team

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -25,14 +25,12 @@ jobs:
           echo "signing.secretKeyRingFile=${HOME}/private.gpg" >> ~/.gradle/gradle.properties
           echo "centralUsername=${{ secrets.CENTRAL_USERNAME }}" >> ~/.gradle/gradle.properties
           echo "centralPassword=${{ secrets.CENTRAL_PASSWORD }}" >> ~/.gradle/gradle.properties
-          echo "sonatypeUsername=${{ secrets.OSSRH_USERNAME }}" >> ~/.gradle/gradle.properties
-          echo "sonatypePassword=${{ secrets.OSSRH_PASSWORD }}" >> ~/.gradle/gradle.properties
-      # deploy to Central Portal with fallback to OSSRH
+      # deploy to Central Portal, throw error if CENTRAL_USERNAME is not set
       - run: |
           if [ -n "${{ secrets.CENTRAL_USERNAME }}" ]; then
             echo "Publishing to Central Portal..."
             ./gradlew publishToCentralPortal
           else
-            echo "Falling back to OSSRH..."
-            ./gradlew publishToSonatype closeAndReleaseSonatypeStagingRepository
+            echo "Error: CENTRAL_USERNAME is not set. Publishing to Central Portal requires this secret."
+            exit 1
           fi

--- a/build.gradle
+++ b/build.gradle
@@ -63,8 +63,8 @@ subprojects {
             url 'https://repo.smeup.cloud/nexus/repository/public/'
         }
         maven { url 'https://jitpack.io' }
-        // the sonatype repo inclusion allows to work with the reload snapshot version
-        maven { url 'https://s01.oss.sonatype.org/content/repositories/snapshots/' }
+        // the repo inclusion allows to work with the reload snapshot version
+        maven { url 'https://central.sonatype.com/repository/maven-snapshots/' }
         maven { url 'https://oss.jfrog.org/oss-snapshot-local/' }
         jcenter()
     }
@@ -91,19 +91,11 @@ nexusPublishing {
     // however take in account that this flag in the other cases must be true or false depending on the version name
     useStaging = !(project.gradle.startParameter.taskNames.contains("publishToSmeup") || jarikoVersion.endsWith("SNAPSHOT"))
     repositories {
-        // Primary: Central Portal (new Sonatype publishing system)
         centralPortal {
             nexusUrl = uri("https://ossrh-staging-api.central.sonatype.com/service/local/")
             snapshotRepositoryUrl = uri("https://central.sonatype.com/repository/maven-snapshots/")
             username = findProperty("centralUsername")
             password = findProperty("centralPassword")
-        }
-        // Fallback: Legacy OSSRH (for backward compatibility)
-        sonatype {
-            nexusUrl = uri("https://s01.oss.sonatype.org/service/local/")
-            snapshotRepositoryUrl = uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
-            username = findProperty("sonatypeUsername")
-            password = findProperty("sonatypePassword")
         }
         smeup {
             nexusUrl = uri("https://repo.smeup.cloud/nexus/content/repositories/releases/")

--- a/docs/development.md
+++ b/docs/development.md
@@ -97,9 +97,9 @@ If you want to force the execution of all checks:
  _try to clean the .gradle directory_)
 
 ## Dependency from develop-SNAPSHOT
-This snapshot is published in Maven Central via Sonatype Central Portal (migrated from deprecated OSSRH), if you want to work with this version you can:
- - cloning jariko repo and to deploy in maven local by using ./gradlew deploy
- - or adding this maven repository url https://s01.oss.sonatype.org/content/repositories/snapshots/ to your pom or gradle script
+This snapshot is published in Maven Central via Sonatype Central Portal (migrated from deprecated OSSRH), if you want to work with SNAPSHOT version you can:
+ - cloning jariko repo and to deploy in maven local by using ./gradlew publishToMavenLocal
+ - or adding this maven repository url https://central.sonatype.com/repository/maven-snapshots/ to your pom or gradle script
 
 ## Tests regarding db native access
 

